catalog/spatial/csw/spatial-csw-source-common/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/common/source/AbstractCswSource.java

```diff
@@ -919,6 +919,11 @@ public ResourceResponse retrieveResource(
       URI resourceUri, Map<String, Serializable> requestProperties)
       throws IOException, ResourceNotFoundException, ResourceNotSupportedException {
 
+    if (resourceUri == null) {
```

Comment:
❓ It would seem below that we can return a response without ever checking this particular parameter. Should we really check it at this point or should we move the check further down after the `if (canRetrieveResourceById())`?